tfx_bsl/tfxio/parquet_tfxio_test.py

```diff
@@ -301,6 +301,53 @@ def _AssertFn(record_batch_list):
       record_batch_pcoll = (p | tfxio.BeamSource(batch_size=_NUM_ROWS))
       beam_testing_util.assert_that(record_batch_pcoll, _AssertFn)
 
+  def testOptionalColumnNames(self):
+    """Tests various valid schemas."""
+    tfxio = ParquetTFXIO(
+        file_pattern=self._example_file,
+        schema=_SCHEMA)
+
+    def _AssertFn(record_batch_list):
+      self.assertLen(record_batch_list, 1)
+      record_batch = record_batch_list[0]
+      self._ValidateRecordBatch(record_batch, _EXPECTED_ARROW_SCHEMA)
+
+    with beam.Pipeline() as p:
+      record_batch_pcoll = (p | tfxio.BeamSource(batch_size=_NUM_ROWS))
+      beam_testing_util.assert_that(record_batch_pcoll, _AssertFn)
+
+  def testOptionalColumnNamesAndSchema(self):
+    """Tests various valid schemas."""
+    tfxio = ParquetTFXIO(file_pattern=self._example_file)
+
+    def _AssertFn(record_batch_list):
+      self.assertLen(record_batch_list, 1)
+      record_batch = record_batch_list[0]
+      self._ValidateRecordBatch(record_batch, _EXPECTED_ARROW_SCHEMA)
+
+    with beam.Pipeline() as p:
+      record_batch_pcoll = (p | tfxio.BeamSource(batch_size=_NUM_ROWS))
+      beam_testing_util.assert_that(record_batch_pcoll, _AssertFn)
+
+  def testSubsetOfColumnNames(self):
+    """Tests various valid schemas."""
+    tfxio = ParquetTFXIO(
+      file_pattern=self._example_file,
+      column_names=['int_feature'],
+      schema=_SCHEMA)
```

Comment:
can this just contain the int_feature?

Reply:
sure, I was trying to test that even when the schema has all fields, but the requested columns are a subset, the result schema and result records will have just a subset

Reply:
I added another test case for this